scripts/hotWalletCli/wallet.ts

```diff
@@ -0,0 +1,336 @@
+
+const BIP32_PATH = `m/44'/931'/0'/0/0`
+const SHAPESHIFT_MULTISIG_ADDRESS = 'thor1xmaggkcln5m5fnha2780xrdrulmplvfrz6wj3l'
+const THORNODE_URL = 'https://daemon.thorchain.shapeshift.com'
+
+const addressNList = bip32ToAddressNList(BIP32_PATH)
+
+type TxsByStakingAddress = Record<string, { signedTx: string; txId: string }>
+
+const suffix = (text: string): string => {
+  return `\n${symbols.error} ${chalk.bold.red(text)}`
+}
+
+export class Wallet {
+  private hdwallet: NativeHDWallet
+
+  constructor(mnemonic: string) {
+    this.hdwallet = new NativeHDWallet({ mnemonic, deviceId: 'hot' })
+  }
+
+  static async new(mnemonic: string): Promise<Wallet> {
+    try {
+      const wallet = new Wallet(mnemonic)
+      const initialized = await wallet.initialize()
+
+      if (!initialized) {
+        error('Failed to initialize hot wallet, exiting.')
+        process.exit(1)
+      }
+
+      const { address, path } = await wallet.getAddress()
+
+      info(`Hot wallet address: ${address} (${path})`)
+
+      return wallet
+    } catch {
+      error('Failed to create hot wallet, exiting.')
+      process.exit(1)
+    }
+  }
+
+  private async initialize(): Promise<boolean | null> {
+    return this.hdwallet.initialize()
+  }
+
+  private async getAddress() {
+    try {
+      const address = await this.hdwallet.thorchainGetAddress({ addressNList })
+
+      if (!address) {
+        error('Failed to get address from hot wallet, exiting.')
+        process.exit(1)
+      }
+
+      return { address, path: BIP32_PATH }
+    } catch {
+      error('Failed to get address from hot wallet, exiting.')
+      process.exit(1)
+    }
+  }
+
+  private async buildFundingTransaction(amount: string, epoch: number) {
+    const { address } = await this.getAddress()
+
+    return {
+      body: {
+        messages: [
+          {
+            '@type': '/types.MsgSend',
+            from_address: SHAPESHIFT_MULTISIG_ADDRESS,
+            to_address: address,
+            amount: [
+              {
+                denom: 'rune',
+                amount,
+              },
+            ],
+          },
+        ],
+        memo: `Fund rFOX rewards distribution - Epoch #${epoch}`,
+        timeout_height: '0',
+        extension_options: [],
+        non_critical_extension_options: [],
+      },
+      auth_info: {
+        signer_infos: [],
+        fee: {
+          amount: [],
+          gas_limit: '0',
+          payer: '',
+          granter: '',
+        },
+      },
+      signatures: [],
+    }
+  }
+
+  async fund(epoch: Epoch) {
+    const { address } = await this.getAddress()
+
+    const distributions = Object.values(epoch.distributionsByStakingAddress)
+
+    const totalDistribution = distributions.reduce((prev, distribution) => {
+      return prev + BigInt(distribution.amount)
+    }, 0n)
+
+    const totalFees = BigInt(2000000) * BigInt(distributions.length)
+
+    const totalAmount = (totalDistribution + totalFees).toString()
```

Comment:
Should we validate this against the published epoch metadata as a sanity check?

Reply:
This is calculated from the published epoch directly. Process is ask for epoch ipfs hash, and use this data to calc distribution. I will have a follow up to update everything to interact with the top level metadata directly, but for all intents and purposes this is validated due to source. Did you have something further in mind?